File: TrendAnalyzerServer/src/main/java/vorotilenko/trendanalyzerserver/server/ServerMessage.java
package vorotilenko.trendanalyzerserver.server;

/**
 * A message which {@link WSServerEndpoint} sends to the client (encoded by JSON)
 */
public class ServerMessage {
    /**
     * One of the {@link ServerMessageTypes}
     */
    public int messageType;
    /**
     * JSON-encoded message wrapped in the {@link ServerMessage}
     */
    public String message;

    public ServerMessage() {

    }

    public ServerMessage(int messageType, String message) {
        this.messageType = messageType;
        this.message = message;
    }
}

File: TrendAnalyzer/app/src/main/java/vorotilenko/trendanalyzer/serverinteraction/ServerMessage.kt
package vorotilenko.trendanalyzer.serverinteraction

/**
 * A message which the server sends to the [WSClientEndpoint] (encoded by JSON).
 */
data class ServerMessage(
    /**
     * One of the [ServerMessageTypes].
     */
    val messageType: Int,
    /**
     * JSON-encoded message wrapped in the [ServerMessage].
     */
    val message: String
)
